src/vs/workbench/services/workingCopy/common/storedFileWorkingCopy.ts

```diff
@@ -1248,7 +1248,7 @@ export class StoredFileWorkingCopy<M extends IStoredFileWorkingCopyModel> extend
 	//#region Utilities
 
 	isReadonly(): boolean {
```

Comment:
I would suggest to make this also `override isReadonly(): boolean | IMarkdownString {` so that for example notebooks can pick this up.

Reply:
And maybe we can adopt this for notebooks as well with this PR.

Reply:
I have started this change, but it doesn't seem that notebook editors handle read-only right now. Ex., if I have an un-checked-out PR that has a notebook change in it I can still modify the notebook editor, even though the file tab has a lock on it.

Reply:
Hm strange, can you try with my readonly mode that I added last milestone? There is also a command to toggle readonly.

Reply:
Your readonly mode does work. I have finished adopting for notebooks, but it's a bit hard to test end to end since readonly+notebooks doesn't work in an extension provided filesystem. I've created this bug: https://github.com/microsoft/vscode/issues/185320